### PR TITLE
Disable performance cops.

### PR DESCRIPTION
**Pull request by michaelstalker**
Our style guide doesn't mention these. It is better to optimize for
readability up front, and tweak performance after profiling the app.

### CODE REVIEW

File: ruby/.rubocop.yml [hunk 1]
```diff
@@ -196,3 +196,67 @@ Lint/UnusedMethodArgument:
...
+                  Place `when` conditions that use splat at the end
+                  of the list of `when` branches.
+  Enabled: false
+
+Performance/Count:
+  Description: >-
+                  Use `count` instead of `select...size`, `reject...size`,
+                  `select...count`, `reject...count`, `select...length`,
+                  and `reject...length`.
+  Enabled: false
```
Comment (mneumark) [inline ruby/.rubocop.yml]: I actually really like this cop. Often times classes do not have `#size` or `#length`. But they always have `#count`.

**Comment by mneumark** [comment]
Well, if the purpose of this is to get it to match our style guide and then we tweak it later, then I guess :+1:  :disappointed:

**Comment by michaelstalker** [comment]
What if we close this PR, and open a new one that:
* disables the cops we don't like
* adds the cops we like to the style guide file

**Comment by mneumark** [comment]
Or I could just commit some additional stuff to this PR as it is and put it on hold.

File: ruby/style_guide.md [hunk 2]
```diff
@@ -1741,20 +1741,126 @@ this rule only to arrays with two or more elements.
...
 
-   # bad - if we just use || operator with falsy value we won't get the expected result
-   batman[:is_evil] || true # => true
+    # bad - if we just use || operator with falsy value we won't get the expected result
+    batman[:is_evil] || true # => true
 
-   # good - fetch work correctly with falsy values
-   batman.fetch(:is_evil, true) # => false
-   ```
+    # good - fetch work correctly with falsy values
```
Comment (pasha) [inline ruby/style_guide.md]: But it is much slower
```ruby
require "benchmark/ips"
require 'active_support'

HASH_WITH_SYMBOL = { fast: "ruby" }
HASH_WITH_STRING = { "fast" => "ruby" }

def fetch_with_default
  HASH_WITH_STRING.fetch("fast", {})
end

def hash_or_default
  HASH_WITH_STRING["fast"] || {}
end

def hash_present_or_default
  HASH_WITH_STRING["fast"].present? || {}
end

def hash_access
  HASH_WITH_STRING["fast"]
end

Benchmark.ips do |x|
  x.report("Hash#[], hash_access") { hash_access }
  x.report("Hash#[], hash_or_default") { hash_or_default }
  x.report("Hash#[], hash_present_or_default") { hash_present_or_default }
  x.report("Hash#fetch, fetch_with_default") { fetch_with_default }
  x.compare!
end

Calculating -------------------------------------
Hash#[], hash_access    26.154k i/100ms
Hash#[], hash_or_default
                        25.585k i/100ms
Hash#[], hash_present_or_default
                        18.488k i/100ms
Hash#fetch, fetch_with_default
                        24.933k i/100ms
-------------------------------------------------
Hash#[], hash_access      3.306M (±18.1%) i/s -     15.562M
Hash#[], hash_or_default
                          3.285M (±18.4%) i/s -     15.428M
Hash#[], hash_present_or_default
                        985.178k (±12.8%) i/s -      4.825M
Hash#fetch, fetch_with_default
                          2.314M (±21.8%) i/s -     10.771M

Comparison:
Hash#[], hash_access:  3305607.5 i/s
Hash#[], hash_or_default:  3285302.4 i/s - 1.01x slower
Hash#fetch, fetch_with_default:  2313501.6 i/s - 1.43x slower
Hash#[], hash_present_or_default:   985178.0 i/s - 3.36x slower
```

Comment (michaelstalker) [inline ruby/style_guide.md]: The change from lines 1744-1750 was just an indentation change. Would you be alright discussing the performance of `#[]` and `#fetch` in another PR?

Comment (colinrymer) [inline ruby/style_guide.md]: To give a fair comparison, you should also include the block version of `#fetch`.

Comment (colinrymer) [inline ruby/style_guide.md]: And `Hash.new({}`.

Comment (michaelstalker) [inline ruby/style_guide.md]: `#fetch` and `#[]` work differently, so we're not comparing apples to apples. `#fetch` won't ever use the default value if the hash key is present. `#[]...||` will use the default value if the hash key is missing, or if its value is falsey.

**Comment by michaelstalker** [comment]
I'm closing this PR, and will separate the disabling of cops from the enabling into separate PRs. The first PR will make `.rubocop.yml` consistent with our current style guide. The following PRs will have @michaelnease's work to each re-enable some of the cops.

**Comment by michaelstalker** [comment]
Correction: That was @mneumark's work. Sorry for the misattribution, Micah! GitHub's autocomplete worked against me.

**Pull request closed without merging**